### PR TITLE
Retriever with score  #11062

### PR DESCRIPTION
**Pull request by vanamayaswanth**

### CODE REVIEW

**Comment by vanamayaswanth** [comment]
This PR will help to resolve the issue #11062  , please review and let me know

**Review by sethusaim — CHANGES_REQUESTED** [review]
Changes look good

**Review by sethusaim — APPROVED** [review]

File: libs/langchain/langchain/schema/vectorstore.py [hunk 1]
```diff
@@ -553,6 +553,7 @@ class VectorStoreRetriever(BaseRetriever):
         "similarity",
         "similarity_score_threshold",
         "mmr",
+        "similarity_score"
```
Comment (eyurtsev) [inline libs/langchain/langchain/schema/vectorstore.py:556]: Let's not introduce a new search type for this purpose -- the names are too similar sounding and will be confusing; e.g., why does similarity_score return a score but not similarity_score_threshold?

Instead, let's add a parameter called `score_key: Optional[str] = None` If specified, it'll place a score when available in the metadata under the specified key. It sounds like this is only supported for now for `similarity_score_threshold` so we should document that clearly and raise a ValueError on an invalid parameter combination.

**Review by eyurtsev — CHANGES_REQUESTED** [review]

**Comment by sethusaim** [comment]
One question what combinations should we allow the user to control ....for example - 
1. The user wants to filter based on the threshold - default option which is already there
2. The user wants to filter based on the threshold and wants the score of returned docs -  currently not sending any score
3. The user wants to get the scores of all the docs without threshold 

Also appending the score in metadata seemed like a better option since it can be treated as unique key like  
how we have for source, etc  [ which the user not explicity change ]. If the user does not want the "score" key, it is fine and he can choose to use it.
Ofcourse we can implement this without having a new search type

Attaching screenshot for point 2

image

**Comment by hwchase17** [comment]
were considering ways to do this in a more native way

**Pull request closed without merging**